org.eclipse.winery.frontends/app/topologymodeler/src/app/redux/reducers/winery.reducer.ts

```diff
@@ -286,32 +286,61 @@ export const WineryReducer =
             case WineryActions.SET_POLICY:
                 const newPolicy: any = (<SetPolicyAction>action).nodePolicy;
                 const policy = newPolicy.newPolicy;
-                const indexOfNodePolicy = lastState.currentJsonTopology.nodeTemplates
-                    .map(node => node.id).indexOf(newPolicy.nodeId);
-                const nodePolicyTemplate = lastState.currentJsonTopology.nodeTemplates
-                    .find(nodeTemplate => nodeTemplate.id === newPolicy.nodeId);
-                const policyExist = nodePolicyTemplate.policies && nodePolicyTemplate.policies.policy;
-
-                return <WineryState>{
-                    ...lastState,
-                    currentJsonTopology: {
-                        ...lastState.currentJsonTopology,
-                        nodeTemplates: lastState.currentJsonTopology.nodeTemplates
-                            .map(nodeTemplate => nodeTemplate.id === newPolicy.nodeId ?
-                                nodeTemplate.generateNewNodeTemplateWithUpdatedAttribute('policies',
-                                    policyExist ? {
-                                        policy: [
-                                            ...lastState.currentJsonTopology.nodeTemplates[indexOfNodePolicy].policies.policy,
-                                            policy
-                                        ]
-                                    } : {
-                                        policy: [
-                                            policy
-                                        ]
-                                    }) : nodeTemplate
-                            )
-                    }
-                };
+                if (lastState.currentJsonTopology.nodeTemplates.includes(newPolicy.nodeId)) {
```

Comment:
Why don't you just use a second action that adds policies to a relationship template?